framework/Models/PostProcessors/Validations/PhysicsGuidedCoverageMapping.py

```diff
@@ -0,0 +1,202 @@
+    """
+    dataDict = {self.getDataSetName(data): data for _, _, data in inputIn['Data']}
+    pivotParameter = self.pivotParameter
+    names = [self.getDataSetName(inp[-1]) for inp in inputIn['Data']]
+    if len(inputIn['Data'][0][-1].indexes) and self.pivotParameter is None:
+      if 'dynamic' not in self.dynamicType:
+        self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataDict, **{'dataobjectNames': names}).items()}
+
+    if pivotParameter:
+      if len(inputIn['Data'][0][-1]['time']) != len(list(evaluation.values())[0]):
+        self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+      if pivotParameter not in evaluation:
+        evaluation[pivotParameter] = inputIn['Data'][0][-1]['time']
+    return evaluation
+
+  ### utility functions
+  def _evaluate(self, datasets, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, datasets, list, list of datasets (data1,data2,etc.) to used.
+      @ In, kwargs, dict, keyword arguments
+      @ Out, outputDict, dict, dictionary containing the results {"feat"_"target"_"metric_name":value}
+    """
+    names = kwargs.get('dataobjectNames')
+    outputDict = {}
+    for feat, targ, msr in zip(self.features, self.targets, self.measurements):
+      featData = self._getDataFromDataDict(datasets, feat, names)
+      targData = self._getDataFromDataDict(datasets, targ, names)
+      msrData = self._getDataFromDataDict(datasets, msr, names)
+
+    # Values from first entry of tuple data from _getDataFromDataDict
+    yExp = np.array(featData[0])
+    yApp = np.array(targData[0])
+    yMsr = np.array(msrData[0])
+    # Sample mean as reference value
+    yExpRef = np.mean(yExp, axis=0)
+    yAppRef = np.mean(yApp)
+    # Standardization
+    yExpStd = (yExp-yExpRef)/yExpRef
+    yAppStd = (yApp-yAppRef)/yAppRef
+    if yExp.shape[1]!= yMsr.shape[1]:
+      self.raiseAnError(IOError, "Number of Measurements is not consistent with number of Experiments/Features.")
+    yMsrStd = (yMsr-yExpRef)/yExpRef
+
+    # Single Experiment response
+    if yExpStd.shape[1]==1:
+      yExpReg = yExpStd.flatten()
+    # Pseudo response of multiple Experiment responses
+    # OrthogonalMatchingPursuit from sklearn used here
+    # Possibly change to other regressors      
+    elif yExpStd.shape[1]>1:
+      regrExp = OrthogonalMatchingPursuit(fit_intercept=False).fit(yExpStd, yAppStd)
+      yExpReg = regrExp.predict(yExpStd)
+    
+    # Kernel Desnity Estimation
+    m1 = yExpReg[:]
+    m2 = yAppStd.flatten()
+    xmin = m1.min()
+    xmax = m1.max()
+    ymin = m2.min()
+    ymax = m2.max()
+
+    binKDE = 200j
```

Comment:
Move this to class __init__ method, or possibly let the user to decide the value

Reply:
Moved to init.

Reply:
Removed.